vars/detectDockerAgent.groovy

```diff
@@ -6,21 +6,50 @@
  * this file be licensed under the Apache-2.0 license or a
  * compatible open source license.
  */
+
+/**@
+ * Contains logic to detect the Docker Agent and Images
+ *
+ * @param args A map of the following parameters
+ * @param args.manifest <optional> The input manifest path, default to "manifests/${INPUT_MANIFEST}"
+ * @param args.distribution <optional> The distribution name, default to "tar"
+ * @param args.platform <optional> The platform name, default to "linux"
+ */
 Map call(Map args = [:]) {
     def lib = library(identifier: "jenkins@main", retriever: legacySCM(scm))
     String manifest = args.manifest ?: "manifests/${INPUT_MANIFEST}"
+    String distribution = args.distribution ?: "tar"
+    String platform = args.platform ?: "linux"
+    echo "manifest: ${manifest}, dist: ${distribution}, plat: ${platform}"
+
     def inputManifest = lib.jenkins.InputManifest.new(readYaml(file: manifest))
-    dockerImage = inputManifest.ci?.image?.name ?: 'opensearchstaging/ci-runner:ci-runner-centos7-v1'
-    dockerArgs = inputManifest.ci?.image?.args
-    // Using default javaVersion as openjdk-17
-    String javaVersion = 'openjdk-17'
+    def schemaVersion = inputManifest.schemaVersion ?: "None"
```

Comment:
is there any other default value we can use for version?

Reply:
Because there is already 1.0 and 1.1 before 1.2, it is just easier to set default as latest.
Therefore in order to avoid hardcoding 1.2, I just leave it as None, so that as long as it is not < 1.2 (non or >=1.2), it will be treated as the latest schemaversion.

Thanks.